### PR TITLE
BUG: Fix CMake options scikit-build-core on macOS and Linux

### PR DESCRIPTION
**Pull request by SimonRit**
This is propagating to MacOS and Linux what's in Windows:
https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/master/scripts/windows_build_module_wheels.py#L99

The Linux version has been tested in https://github.com/RTKConsortium/ITKCudaCommon/actions/runs/9275896116/job/25521657471. I haven't tested MacOS' version.

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]
LGTM, but some better versed in bash should review too.

**Comment by SimonRit** [comment]
I have a bug on Windows (see [here](https://github.com/RTKConsortium/ITKCudaCommon/actions/runs/9276277284/job/25538257384), the string is not adequately handled) so you may want to wait the patch before merging

**Comment by SimonRit** [comment]
> I have a bug on Windows (see [here](https://github.com/RTKConsortium/ITKCudaCommon/actions/runs/9276277284/job/25538257384), the string is not adequately handled) so you may want to wait the patch before merging

No sorry, the problem is not in ITKPythonPackage I believe, this PR can be merged.

**Review by thewtex — APPROVED** [review]
@SimonRit thank you!

**Pull request merged**
